graylog-storage-elasticsearch7/src/main/java/org/graylog/storage/elasticsearch7/IndicesAdapterES7.java

```diff
@@ -417,8 +422,13 @@ public void optimizeIndex(String index, int maxNumSegments, Duration timeout) {
                 .indices(index)
                 .maxNumSegments(maxNumSegments)
                 .flush(true);
-
-        client.execute((c, requestOptions) -> c.indices().forcemerge(request, withTimeout(requestOptions, timeout)));
+        try {
+            final FutureActionListener<ForceMergeResponse> future = new FutureActionListener<>();
+            client.execute((c, requestOptions) -> c.indices().forcemergeAsync(request, withTimeout(requestOptions, timeout), future));
+            future.getBlocking(timeout);
```

Comment:
I'm sorry, I may have not understood everything correctly, but what is the point of introducing an async call if in the next line you block the execution with getBlocking() ? I'm afraid it won't change anything, the thread executing optimizeIndex() method will have to wait for the completion anyway...